### PR TITLE
Add language guide and entry in table for Dart

### PR DESCRIPTION
**Pull request by mvxt**
# Description
Added an entry to the `tutorials` table and a language guide for Dart.

# Reasons
Adding more examples and more languages/frameworks, building out our examples to prove the concept that CircleCI can support various use cases

### CODE REVIEW

File: jekyll/_cci2/language-dart.md [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+order: 2
+---
+
+This document is a walkthrough of a sample Dart project setup on CircleCI. The project cherry-picks `number_guesser` and `number_thinker` code from [Write HTTP clients & servers](https://dart.dev/tutorials/server/httpserver) tutorial.
+
+We assume you have a basic working knowledge of CircleCI and associated terminology. If not, visit our [Getting Started docs](http://0.0.0.0:4000/docs/2.0/getting-started/#section=getting-started).
+
+## Quickstart
+
+**Repository**: <https://github.com/CircleCI-Public/circleci-dart-demo>
```
Comment (teesloane) [inline jekyll/_cci2/language-dart.md:16]: Can you make these into markdown formatted links?

File: jekyll/_cci2/language-dart.md [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+**Repository**: <https://github.com/CircleCI-Public/circleci-dart-demo>   
+**Builds**: <https://app.circleci.com/pipelines/github/CircleCI-Public/circleci-dart-demo>
+
+1. Find the full configuration at the bottom of this document or in the repository linked above.
+1. Copy that config into [`.circleci/config.yml`]({{ site.baseurl }}/2.0/configuration-reference/) in your project's root directory.
+1. Modify as needed.
+
+The CircleCI config does the following:
+
+- **It runs the tests on a Docker container.**
```
Comment (teesloane) [inline jekyll/_cci2/language-dart.md:25]:
```suggestion
- **It runs the tests in a Docker container.**
```

File: jekyll/_cci2/language-dart.md [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+    - The `test` job uses the `google/dart` Docker image as its execution image (CircleCI doesn't currently have a native Dart image).
```
Comment (teesloane) [inline jekyll/_cci2/language-dart.md:26]:
```suggestion
    - The `test` job uses the `google/dart` Docker image as its execution image.
```
I think it's unnecessary to add that we do not have an image.

File: jekyll/_cci2/language-dart.md [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+    - The tests uses the [junitreporter](https://pub.dev/packages/junitreport) package to produce JUnit XML output for CircleCI's [test metadata feature](https://circleci.com/docs/2.0/collect-test-data/), which in turn supports things like test summary, intelligent test splitting, and insights data/metrics.
```
Comment (teesloane) [inline jekyll/_cci2/language-dart.md:27]:
```suggestion
    - The tests use the [junitreporter](https://pub.dev/packages/junitreport) package to produce JUnit XML output for CircleCI's [test metadata feature](https://circleci.com/docs/2.0/collect-test-data/), which in turn supports features such as test summary, intelligent test splitting, and insights data/metrics.
```

File: jekyll/_cci2/language-dart.md [hunk 5]
```diff
@@ -0,0 +1,435 @@
+---
+layout: classic-docs
+title: "Language Guide: Dart"
+short-title: "Dart"
+description: "Overview and sample config for a Dart project"
+categories: [language-guides]
+order: 2
+---
+
+This document is a walkthrough of a sample Dart project setup on CircleCI. The project cherry-picks `number_guesser` and `number_thinker` code from [Write HTTP clients & servers](https://dart.dev/tutorials/server/httpserver) tutorial.
```
Comment (teesloane) [inline jekyll/_cci2/language-dart.md:10]:
```suggestion
This document is a walkthrough of a sample Dart project setup on CircleCI. The project cherry-picks `number_guesser` and `number_thinker` code from the [Write HTTP clients & servers](https://dart.dev/tutorials/server/httpserver) tutorial.
```

File: jekyll/_cci2/language-dart.md [hunk 6]
```diff
@@ -0,0 +1,435 @@
...
+#####################
+# Common Definitions
+#####################
+
+# Orb declarations
+orbs:
+  win: circleci/windows@2.4.0
+```
+
+- **Config Version**: Current is 2.1. Unless using legacy server install or legacy jobs and workflows, we highly suggest using 2.1.
```
Comment (teesloane) [inline jekyll/_cci2/language-dart.md:63]:
```suggestion
- **Config Version**: Current is `2.1`. Unless using a legacy server installation,  jobs or workflows, we recommend using 2.1.
```

File: jekyll/_cci2/language-dart.md [hunk 7]
```diff
@@ -0,0 +1,435 @@
...
+
+{% endraw %}
+
+- CircleCI offers [reusable configuration syntax]({{ site.baseurl }}/2.0/reusing-config), which is what our orbs are built off of. Under `commands`, we define a few parameterized commands that run the same sequence of steps.
+    - `dependencies` checks out the code and restores/saves a dependency cache (Dart packages downloaded from last build).
+    - `native-build` uses the `dart2native` command to compile a native executable for each operating system and then artifacts it, making it available to download from the job details.
+    - In both commands, we take parameters to override a cache path and shell for different jobs. Dart's package cache is in a different location in Windows, and it also uses a different shell (Powershell).
+
+```yaml
+######################
```
Comment (teesloane) [inline jekyll/_cci2/language-dart.md:122]: I think I would recommend removing the large comment blocks - the sections are already broken up, and none of our language guides use them.

File: jekyll/_cci2/language-dart.md [hunk 8]
```diff
@@ -0,0 +1,435 @@
...
+      - build-docker:
+          # Uncomment/modify context to use secrets (e.g., DockerHub credentials, etc.)
+          #context: dart-docker
+          requires:
+            - test
+```
+
+- In the `workflows` section, we define a workflow called `test-and-build` and define which jobs are run.
+- Each of the `build-*` jobs depend on `test` passing, which is set by the `requires` key.
+- There's a commented line for `build-docker`, which is using a `dart-docker` context. See links to resources at the bottom of this document.
```
Comment (teesloane) [inline jekyll/_cci2/language-dart.md:148]:
```suggestion
- The commented line for `build-docker` uses a `dart-docker` context. See links to resources at the bottom of this document.
```

**Comment by mvxt** [comment]
@teesloane Latest changes pushed! Let me know if there's more

**Pull request merged**